master/buildbot/worker/kubernetes.py

```diff
@@ -0,0 +1,303 @@
+    )
+
+    def __init__(self, kube_obj):
+        self.kube_obj = kube_obj
+
+    @defer.inlineCallbacks
+    def getRenderingFor(self, props):
+        res = yield self.recursive_render(
+            copy.deepcopy(self.kube_obj),
+            props
+        )
+        defer.returnValue(res)
+
+    @defer.inlineCallbacks
+    def recursive_render(self, obj, props):
+        """Recursively parse kubernetes object tree to find renderable"""
+        # This code is inspired by the code of kubernetes client-python
+        # https://github.com/kubernetes-incubator/client-python/blob/4e593a7530a8751c817cceec715bfe1d03997793/kubernetes/client/api_client.py#L172-L214
+        if isinstance(obj, string_types + (text_type,)):
+            res = yield Interpolate(obj).getRenderingFor(props)
+        elif isinstance(obj, tuple):
+            res = yield self.recursive_render(list(obj), props)
+        elif isinstance(obj, list):
+            res = []
+            for sub_obj in obj:
+                temp = yield self.recursive_render(sub_obj, props)
+                res.append(temp)
+        elif isinstance(obj, dict):
+            res = {}  # pylint: disable=redefined-variable-type
+            for key, sub_obj in obj.item():
+                res[key] = yield self.recursive_render(sub_obj, props)
+        elif isinstance(obj, IRenderable):
+            res = yield obj.getRenderingFor(props)
+        elif isinstance(obj, self.untouched_types):
+            res = obj
+        else:
+            for key in obj.swagger_types:
+                value = getattr(obj, key)
+                if not value:
+                    continue
+                res = yield self.recursive_render(value, props)
+                setattr(obj, key, res)
+            res = obj
+        defer.returnValue(res)
+
+
+class KubeLatentWorker(AbstractLatentWorker):
+    instance = None
+
+    properties_source = 'kube Latent Worker'
+
+    @staticmethod
+    def dependency_error():
+        config.error("The python module 'kubernetes>=1' is needed to use a "
+                     "KubeLatentWorker")
+
+    def load_config(self, kubeConfig):
+        try:
+            kube_config.load_kube_config()
+        except (FileNotFoundError,
+                kube_config.config_exception.ConfigException):
+            try:
+                kube_config.load_incluster_config()
+            except kube_config.config_exception.ConfigException:
+                pass
+        kubeConfig = kubeConfig or {}
+        for config_key, value in kubeConfig.items():
+            setattr(client.configuration, config_key, value)
+        if not client.configuration.host:
+            config.error("No kube-apimaster host provided")
+
+    @classmethod
+    def default_job(cls):
+        if not client:
+            cls.dependency_error()
+        job_name = '%(prop:buildername)s-%(prop:buildnumber)s'
+        return client.V1Job(
+            metadata=client.V1ObjectMeta(name=job_name),
```

Comment:
For now, I have a couple of call : `create_namespace_job` and `delete_namespace_job.`
But this call create job and job created pods (docker containers). That's how kubernetes handle short live container
That's also why I miss the `stop_container(fast=False)` and `followStartupLogs.` I need to find pod from a job.
This would use `list_namespaced_pod` (https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/docs/CoreV1Api.md#list_namespaced_pod) (with long polling to wait for pod creation or deletion).
'FollowStartupLog' needs `read_namespaced_pod_log` (https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/docs/CoreV1Api.md#read_namespaced_pod_log) with streaming capabilities.
I don't have any other feature in minds (may be managing password with 'secrets'), but each would add a bunch of call. This is basically redevelopping the python client.

But again, I agree the render thing is really ugly. I've read somewere that the python client can also take yaml string (need to figure out how). The string would be renderable by default, without the hassle of developping each call.